Clarify verify docs

smith opened the pull request:
These weren't very clear, so added some content to flesh this out and
remove examples that weren't very helpful.

Please fix any bad writing and make this be duplicated in sections it
needs to be.

Signed-off-by: Nathan L Smith <smith@chef.io>

jamescott commented:
Thanks! Also, gonna make a slight, slight edit right after this :)

jamescott commented:
https://github.com/chef/chef-docs/commit/581234571444d0ec26ec0b96ba4099254a802196

Commit:
Clarify verify docs

These weren't very clear, so added some content to flesh this out and
remove examples that weren't very helpful.

Please fix any bad writing and make this be duplicated in sections it
needs to be.

Signed-off-by: Nathan L Smith <smith@chef.io>

thommay commented on chef_master/source/resource_file.rst:
```diff
@@ -259,9 +259,11 @@ This resource has the following properties:
    **Ruby Types:** String, Block
 
-   A block or a string that returns ``true`` or ``false``. A string, when ``true`` is executed as a system command.
+   Allows verification of a file's contents before it is created. Creates a temp file and then allows execution of commands or Ruby code to make sure the file meets the needed structure.
```
"a temporary file" seems better here.

smith replied:
@thommay updated this a little in 5e87a4c.

thommay commented on chef_master/source/resource_file.rst:
```diff
@@ -259,9 +259,11 @@ This resource has the following properties:
    **Ruby Types:** String, Block
 
-   A block or a string that returns ``true`` or ``false``. A string, when ``true`` is executed as a system command.
+   Allows verification of a file's contents before it is created. Creates a temp file and then allows execution of commands or Ruby code to make sure the file meets the needed structure.
```
"fits the needed structure" doesn't seem to quite capture the intention here, but I'm struggling to come up with better words.
"to ensure the new file is fit for purpose"? I dunno that that's any better.
"the new file is valid"?

thommay reviewed (commented):
This is definitely a huge step forward, thank you!

Commit:
Minor copyedits

Signed-off-by: David Wrede <dwrede@chef.io>

davidwrede commented:
![gif-keyboard-15326130058755744189](https://cloud.githubusercontent.com/assets/1634752/23817412/c704989c-05a7-11e7-89eb-eeb6c4a15db8.gif)
